### PR TITLE
chartsV9 tasks and bug fixes

### PR DESCRIPTION
**Pull request by Anush2303**
## Related Issue(s)

- Fixes #

### CODE REVIEW

File: packages/react-components/react-charts-preview/library/src/components/CommonComponents/useCartesianChartStyles.styles.ts [hunk 1]
```diff
@@ -200,6 +202,7 @@ export const useCartesianChartStyles_unstable = (props: ICartesianChartProps): I
     axisTitle: mergeClasses(cartesianchartClassNames.axisTitle, baseStyles.axisTitle /*props.styles?.axisTitle*/),
     xAxis: mergeClasses(cartesianchartClassNames.xAxis, baseStyles.xAxis /*props.styles?.xAxis*/),
     yAxis: mergeClasses(cartesianchartClassNames.yAxis, baseStyles.yAxis /*props.styles?.yAxis*/),
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/library/src/components/CommonComponents/useCartesianChartStyles.styles.ts:204]: merge yaxisg with yaxis only. Dont create a separate class for it.

File: packages/react-components/react-charts-preview/library/src/components/CommonComponents/useCartesianChartStyles.styles.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ import { HighContrastSelectorBlack, HighContrastSelector } from '../../utilities
 import { SlotClassNames } from '@fluentui/react-utilities/src/index';
 import { tokens } from '@fluentui/react-theme';
 import { isRtl } from '../../utilities/utilities';
+import { merge } from 'd3-array';
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/library/src/components/CommonComponents/useCartesianChartStyles.styles.ts:7]: why is this needed?

File: packages/react-components/react-charts-preview/library/src/types/IDataPoint.ts [hunk 3]
```diff
@@ -134,6 +134,8 @@ export interface IChartDataPoint {
    */
   color?: string;
 
+  gradient?: [string, string];
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/library/src/types/IDataPoint.ts:137]: revert this change

File: packages/react-components/react-charts-preview/stories/src/DonutChart/DonutChartCustomCallout.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import {
+  DonutChart,
+  IChartProps,
+  IChartDataPoint,
+  DataVizPalette,
+  getColorFromToken,
+} from '@fluentui/react-charts-preview';
+import { getGradientFromToken, DataVizGradientPalette } from '@fluentui/react-charting';
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/stories/src/DonutChart/DonutChartCustomCallout.stories.tsx:9]: no references from v8 package

File: packages/react-components/react-charts-preview/stories/src/DonutChart/DonutChartCustomCallout.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+import PopoverComponent from '../../../library/src/components/CommonComponents/Popover';
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/stories/src/DonutChart/DonutChartCustomCallout.stories.tsx:10]: Import dependencies from @fluentui/react-charts-preview

File: packages/react-components/react-charts-preview/stories/src/HorizontalBarChart/HorizontalBarChartCustomCallout.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,153 @@
+import * as React from 'react';
+import {
+  HorizontalBarChart,
+  IChartProps,
+  IChartDataPoint,
+  DataVizPalette,
+  getColorFromToken,
+} from '@fluentui/react-charts-preview';
+import { DirectionalHint } from '@fluentui/react';
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/stories/src/HorizontalBarChart/HorizontalBarChartCustomCallout.stories.tsx:9]: no v8 imports should be added

File: packages/react-components/react-charts-preview/stories/src/VerticalBarChart/VerticalBarChartAxisTooltip.stories.tsx [hunk 7]
```diff
@@ -1,7 +1,17 @@
...
+  Field,
+  Radio,
+  RadioGroup,
+  CheckboxOnChangeData,
+  CheckboxProps,
+  Input,
+  InputProps,
+  InputOnChangeData,
+} from '@fluentui/react-components';
+import { StackShim } from '@fluentui/react-migration-v8-v9';
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/stories/src/VerticalBarChart/VerticalBarChartAxisTooltip.stories.tsx:14]: Use strictly v9 controls only

File: packages/react-components/react-charts-preview/library/src/components/CommonComponents/useCartesianChartStyles.styles.ts (hunk 1 above)
Comment (Anush2303) [reply packages/react-components/react-charts-preview/library/src/components/CommonComponents/useCartesianChartStyles.styles.ts:204]: Okay

**Comment by AtishayMsft** [comment]
Confirm if the build is passing after your change.

File: packages/react-components/react-charts-preview/library/src/components/CommonComponents/useCartesianChartStyles.styles.ts (hunk 2 above)
Comment (Anush2303) [reply packages/react-components/react-charts-preview/library/src/components/CommonComponents/useCartesianChartStyles.styles.ts:7]: Will remove this

File: packages/react-components/react-charts-preview/stories/src/VerticalBarChart/VerticalBarChartAxisTooltip.stories.tsx (hunk 7 above)
Comment (Anush2303) [reply packages/react-components/react-charts-preview/stories/src/VerticalBarChart/VerticalBarChartAxisTooltip.stories.tsx:14]: In the components mapping, it is written Stackshim can be used in V9 instead of Stack in V8

File: packages/react-components/react-charts-preview/library/src/components/CommonComponents/CartesianChart.types.ts [hunk 8]
```diff
@@ -75,6 +75,16 @@ export interface ICartesianChartStyles {
...
 
+  /**
+   * Style for the element containing the y-axis.
+   */
+  yAxisg?: string;
+
+  /**
+   * Style for the rtl mode in yAxis.
+   */
+  rtl?: string;
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/library/src/components/CommonComponents/CartesianChart.types.ts:86]: why do you need to export rtl classname using interface

File: packages/react-components/react-charts-preview/stories/src/VerticalBarChart/VerticalBarChartAxisTooltip.stories.tsx (hunk 7 above)
Comment (AtishayMsft) [reply packages/react-components/react-charts-preview/stories/src/VerticalBarChart/VerticalBarChartAxisTooltip.stories.tsx:14]: dont use stack. Use css grid layout or flex box if needed.
https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_grid_layout/Basic_concepts_of_grid_layout

**Comment by Anush2303** [comment]
> Confirm if the build is passing after your change.

Build is passing.

File: packages/react-components/react-charts-preview/stories/src/LineChart/LineChartCustomLocaleDateAxis.stories.tsx [hunk 9]
```diff
@@ -3,6 +3,10 @@ import { ILineChartProps, LineChart, DataVizPalette } from '@fluentui/react-char
 import { Switch } from '@fluentui/react-components';
 
 export const LCCustomLocale = (props: ILineChartProps) => {
+  React.useEffect(() => {
+    _getCustomLocale();
+  }, []);
+
   const [width, setWidth] = React.useState<number>(700);
   const [height, setHeight] = React.useState<number>(300);
   const [allowMultipleShapes, setAllowMultipleShapes] = React.useState<boolean>(false);
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/stories/src/LineChart/LineChartCustomLocaleDateAxis.stories.tsx:12]: use the default locale in the useState hook directly. useEffect should not be needed

Comment (Anush2303) [reply packages/react-components/react-charts-preview/stories/src/LineChart/LineChartCustomLocaleDateAxis.stories.tsx:12]: Okay

**Review by AtishayMsft — APPROVED** [review]
